odoo/addons/base/models/res_partner.py

```diff
@@ -644,6 +644,7 @@ def write(self, vals):
 
     @api.model_create_multi
     def create(self, vals_list):
+        vals_list = [{k: v.replace('@','') if k == 'name' else v for k, v in vals.items()} for vals in vals_list]
```

Comment:
is it needed since we change the name in `_send_prepare_values` ?

Reply:
1 sec, i have to investigate deeper

Reply:
@std-odoo yeah, i thought about it and i don't think it's possible to pass test without modifying create function.
@tde-banana-odoo

Reply:
the only hope i found was to change [email_split](https://github.com/odoo/odoo/blob/922b4ba13d679a237bcc43d7296ff402beacb022/addons/mail/models/mail_template.py#L169) function, and i don't think if changing that it's even possible. maybe you have some directions so that i can research it further

Reply:
or if u have other ideas we can talk about it, one of them is to modify/create new one `email_split` function and pass names to the templates (i think this one is little bit hacky, cuz email templates should only get emails, but passing the names should work too).

Reply:
hum, looks like you changed the create method of res.partner to fix the tests...

What about doing that in `assertSentEmail` ? 
```python
from odoo.tools import email_split_tuples
email_to_list = [formataddr((name.replace('@', '').strip("' "), email)) for name, email in email_split_tuples(email_to_list)]
expected['email_to'] = email_to_list

# fetch mail
sent_mail = self._find_sent_email(
    expected['email_from'],
    expected['email_to'],
    values.get('subject'),
)
```
(note that we should find something else for the strip, but you got the idea)

Reply:
as a general advice, if the tests break because of a change we introduced, we adapt the tests, we never do hack in the base code to keep the tests green